com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/internal/spi/SpiFrameHandler.java

```diff
@@ -603,32 +599,10 @@ private synchronized void stopRetryTimer() {
     private class RetryTimer extends TimerTask {
         @Override
         public void run() {
-            synchronized (outputFrameSynchronisation) {
-                logger.debug("SPI Timer task [{}].  {}", retries, frameToString(lastFrameSent));
-
-                // Resend the last message sent if there is one
-                if (lastFrameSent == null) {
-                    return;
-                }
+            logger.debug("SPI Timer task triggered.");
 
-                if (retries++ > ACK_TIMEOUTS) {
-                    // Too many retries.
-                    // We should alert the upper layer so they can reset the link?
-                    frameHandler.handleLinkStateChange(false);
-
-                    logger.debug("Error: number of retries exceeded [{}].", retries);
-                    // drop message
-                    lastFrameSent = null;
-                    retries = 0;
-                    return;
-                }
-
-                try {
-                    outputFrame(lastFrameSent);
-                } catch (Exception e) {
-                    logger.warn("Caught exception while attempting to retry message in SpiRetryTimer", e);
-                }
-            }
+            // We should alert the upper layer so they can reset the link
+            frameHandler.handleLinkStateChange(false);
```

Comment:
I fear that the restart here will not work. When the problem from https://github.com/zsmartsystems/com.zsmartsystems.zigbee/issues/604 occurs, we are still initializing.

This means that in

https://github.com/zsmartsystems/com.zsmartsystems.zigbee/blob/5825dcff62edc75e02ced49168d07c1b3078d447/com.zsmartsystems.zigbee.dongle.ember/src/main/java/com/zsmartsystems/zigbee/dongle/ember/ZigBeeDongleEzsp.java#L720-L722

the variable `initialised` is still `false` and thus we will `return` instead of setting the transport to `OFFLINE`.